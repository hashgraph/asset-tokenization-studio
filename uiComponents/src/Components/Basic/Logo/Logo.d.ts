import type { CSSProperties } from "react";
import type { SystemCSSProperties as ChakraSystemCSSProperties } from "@chakra-ui/system";
export type LogoSize = "full" | "iso";
export declare const logoPartsList: Array<"fullImage" | "isoImage">;
type Parts = typeof logoPartsList;
export interface LogoThemeConfiguration {
    baseStyle?: Record<Parts[number], any>;
    variants?: Record<Parts[number], any>;
}
export type LogoThemeStyle = {
    fullImage: string;
    isoImage: string;
};
export interface LogoProps {
    size?: LogoSize;
    variant?: string;
    width?: ChakraSystemCSSProperties["width"];
    height?: ChakraSystemCSSProperties["height"];
    alt: string;
    customStyle?: CSSProperties;
}
export declare const Logo: ({ size, variant, width, height, customStyle, alt, }: LogoProps) => JSX.Element;
export {};
