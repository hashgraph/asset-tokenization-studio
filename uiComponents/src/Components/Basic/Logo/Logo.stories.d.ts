declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./Logo").LogoProps>;
export default meta;
export declare const Full: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Logo").LogoProps>;
export declare const Iso: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Logo").LogoProps>;
