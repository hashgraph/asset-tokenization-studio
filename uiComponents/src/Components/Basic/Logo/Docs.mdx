import {
  Story,
  Meta,
  Title,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as LogoStories from "./Logo.stories";

<Meta of={LogoStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=4026%3A44299&mode=design&t=KlerdyEMWSQWLAVX-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

Add the following configurations to the `theme.tsx` and `Logo.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Logo: ConfigLogo,
  },
};
```

```tsx dark
const ConfigLogo = {
  baseStyle: {
    isoImage: logoIso,
    fullImage: logoFull,
  },
};
```

## Props

<ArgTypes of={LogoStories} />

## Stories

<Stories title="" />
