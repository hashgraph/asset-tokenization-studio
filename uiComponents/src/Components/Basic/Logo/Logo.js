var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
export var logoPartsList = [
    "fullImage",
    "isoImage",
];
export var Logo = function (_a) {
    var _b = _a.size, size = _b === void 0 ? "full" : _b, variant = _a.variant, _c = _a.width, width = _c === void 0 ? "auto" : _c, _d = _a.height, height = _d === void 0 ? "auto" : _d, customStyle = _a.customStyle, alt = _a.alt;
    var _e = useChakraMultiStyleConfig("Logo", {
        variant: variant
    }), fullImage = _e.fullImage, isoImage = _e.isoImage;
    var srcBySize = {
        full: fullImage,
        iso: isoImage
    };
    return (React.createElement("img", { alt: alt, src: srcBySize[size], style: __assign({ width: width, height: height }, customStyle) }));
};
