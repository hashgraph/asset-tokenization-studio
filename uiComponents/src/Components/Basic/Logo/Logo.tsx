import type { CSSProperties } from "react";
import React from "react";
import type { SystemCSSProperties as ChakraSystemCSSProperties } from "@chakra-ui/system";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";

export type LogoSize = "full" | "iso";

export const logoPartsList: Array<"fullImage" | "isoImage"> = [
  "fullImage",
  "isoImage",
];

type Parts = typeof logoPartsList;

export interface LogoThemeConfiguration {
  baseStyle?: Record<Parts[number], any>;
  variants?: Record<Parts[number], any>;
}

export type LogoThemeStyle = {
  fullImage: string;
  isoImage: string;
};

export interface LogoProps {
  size?: LogoSize;
  variant?: string;
  width?: ChakraSystemCSSProperties["width"];
  height?: ChakraSystemCSSProperties["height"];
  alt: string;
  customStyle?: CSSProperties;
}

export const Logo = ({
  size = "full",
  variant,
  width = "auto",
  height = "auto",
  customStyle,
  alt,
}: LogoProps) => {
  const { fullImage, isoImage } = useChakraMultiStyleConfig("Logo", {
    variant,
  }) as LogoThemeStyle;
  const srcBySize: { [key in LogoSize]: string } = {
    full: fullImage,
    iso: isoImage,
  };

  return (
    <img
      alt={alt}
      src={srcBySize[size]}
      style={{ width, height, ...customStyle }}
    />
  );
};
