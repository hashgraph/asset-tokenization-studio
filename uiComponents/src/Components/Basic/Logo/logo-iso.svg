<svg width="56" height="32" viewBox="0 0 56 32" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="56" height="32" fill="url(#pattern0)"/>
<defs>
<pattern id="pattern0" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_1826_13790" transform="matrix(0.00134228 0 0 0.00234899 0 0.0677852)"/>
</pattern>
<image id="image0_1826_13790" width="745" height="368" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAukAAAFwCAYAAAAFReUdAAAACXBIWXMAACE4AAAhOAFFljFgAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAB/DSURBVHgB7d1tkltFti7glfrgENHcaB3s8xv1BGz1CChGgBkBZgSYEWBGgD0CihHgHoGLESDfCXT1b2yOOu69EVxUpTyZqipjA/7C2tKW8nkiqlQujHFpa6N3r71yZQoAaNDo+uwo5fwwYM+tBumz8x/nx8FBGQQAANArQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9Mxpdn92NBqQUD5Y/zucBAAA9N0o5fxkNWKV0Wh6EdAAAek+7CwAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DOjAAAO2vA8/e3nxfw0OEjja7PjwbWb38TBS8fLJ/PPohEq6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9I6QDAEDPCOkAANAzQjoAAPSMkA4AAD0jpAMAQM8I6QAA0DNCOgAA9IyQDgAAPSOkAwBAzwjpAADQM0I6AAD0jJAOAAA9M8opTqIBg/PVJF2fHcWBSykWyx/n8wAAYG+Nzh4/+igaML5+82HK+es4dDkdl8+fBQAAe2sUAC8ymU3eLZ/PxzHJ+eIjBuXj0uC8fO+ZX79IuWN3evV1yhdf17s+w2Usfo5YxGK+CADgKSEdWnQZvpfDmNXQXYLzdJDzB/Wf5CjfuzAt8TrO61erEqrj4qN+/VRah+5X+sPfU753Pix3uerX127Wz6flz1uU7y/KH3u6SulfNdzXUK+NC4DWCOlwyEoYHw7jaFBCeKTVNEX64CKE52kN3+uV45eh+zWydtemV3+J+pBK2f5puC+P4xrkU8xriC/feBR5cFqq+POzs/I9lXgADoyQDgfi3clsWivjg1gd1TBesnd5zBetKOniUw+C+NvJV1X+dFQTfA3x41KNL5X40/L5NKf0fcn28/PzOBHcAdhnQjrsqfF/zWZxXoJ4yjdLVj06L9Xxi5mqF2E8RVOm9aNU34/qzz24DO7l63mJ8v8otwzm2mUA2CdCOuyJWik/H8StEso/rFXyWJUqeepFm0pfTfNFX/2t2tIzfia0Ly8q7acBAD0lpEOPja7PjlJe3Sp18Y9rpbx+r8Eq+aY8De3rFpnrN+eR8/c5DR6cPZ6fBAD0iJAOfTKZTUajmA1y/rQUf2+lXHvKRfJOrPvb06w8x5+PL3raT3JK3wrsAPSBkA49UCvmF8E8l2C+HoMomm/XtHzcLoH9tsAOQB8I6bAj48lslgarj1cp3akVc8G8N6bxTGAvj8e/rAbf6mEHYJuEdNimq7nlkT+PyEe5xHLBvNempaJ+dzzMd9O1mw/OI317/mT+IACgY0I6bEMJ5+8MVp+vUr5TQvkk2DvlTsetcnF1a7Buh0lfmRADQJeEdOhQbWmJ4epOjvxxqZpPVM0PwrRE9m9Gw1jU6vryPH0lrAOwaYMANq4uBB1fu/kwhvmHUnX9VPX88Fwe09vjYf5nOdbflLsl0wCADRHSYYOuwnnK+WHUDYdohbAOwEYJ6bAJJZgJ54SwDsCGCOnwNiazyfj9G1/XYBbCOb9ah/V33r/xpbAOwJ8hpMOfVAPYqIbzlO4E/IHL8Y0Px9dmtwMA3oCQDm/osu/8nzWAWRDKa5jWaTD1NaOqDsDrEtLhddXWlms3ji/7zqcBb2a67le/fuPr+loKAHgJIR1ew/j9G5+vW1sifRrwNnK6U8L6D1pgAHgZIR1e5nJqS6R0T2sLGzS9bIExBQaAPySkwwu8c212a7TejMjUFjpTp8A8rOscAgCeIaTDb12OVcyRv1M9ZwumdZ3DelwjAFwS0uFZtb2lVs+NVWTL1uMar9/8QfsLAJWQDpfG1258etneMg3YhRyz2v4yvDa7FQA0TUiHWE9v+ToiHWtvoQemg8jfaX8BaNsooGW1/3yYvwuLQ+mZi/aXG5Pl2eCrWMwXAUBTVNJp11X/uYBOX13OVNenDtAeIZ02TWYz/efsiXox+VBQB2iLkE5zLheIPtR/zh5Z3/UZ/9dsFgA0QUinKeP3b3xugSh7ahKr/HB8bfZpAHDwhHSaMbp+48tI6V7A/ioXl/lYUAc4fEI6TagBPeV0N+Ag5OP/eH9mwy2AAyakc/AEdA7RKuWvBXWAwyWkc9AEdA5ZDepaXwAOk5DOwRLQaYMedYBDJKRzkOoUFwGdduTj0fXZUQBwMIR0Ds4712a3THGhNSnn78xRBzgcQjqHZTKbriJ/E9CeOkf9OzuTAhwGIZ3DUcLJ2E6itG19DpRzwTkAsOeEdA5DCSXrcFJCSkDbalD/LgDYa0I6B2E8WH0ZAjpcORpfv/F1ALC3hHT2Xh21GCnZ1AWeldMdoxkB9peQzl6rY+eMWoQXyfcsJAXYT0I6+6uEj5RNcoGXmFhICrCfhHT2VgkfNaBPA3iZ6Xi0XrMBwB4R0tlL6z70iKMAXi2nO//x/sy6DYA9IqSzf9ZtLvrQ4U2sUv5SfzrA/hDS2TuX89CBNzO5bBEDYA+MAvbIus0l60PfppxjUS7nT1OOeUT6d3n+T1fDWJRfn6azWAyj/POXOBtdHK+cYjo4j0mUxxz5g/pY/oxZsE1Hte3l//80vxcA9JqQzv64mOZyN+hUCdMnKadHq0HMx8s4+XkxP33Z71/GK51efXH+B/9wPJnNSuifltt6RznlDwX3bl22vTyIVxxXAHZLSGdvaHPpRq2Up5S+LeH8wXtnMV8s5k8r4+fRveViXir0MS//rQf115PJbPJ/hnFUQvut8pf7sFbcg01at72Ui6uPAoDeEtLZC9pcNuvZYH72ZH5y9f1F7N7lRcKDq9B+sWFV3BbYN0rbC0DPCen0nzaXjamtLBHpq2eDed+dPV7/XetHjK/NbueUPy2h/Sh4K5dtL8fxzJ0TAPrDdBd6bzxc3Q3eUjrOKX109vjRR5ehdy8tn8yP688Q5+nv5Wf6NngbE5scAfSXkE6v1cppCWOfBn9SOh6ep7+VcPvZPofz36p97OVnul1/NmH9LeR0p7YTBQC9I6TTc1ml709Yt7WUanMN5z8f8BSP+rPVsL6urOdfp8jw+lJ2jgH0kZBOb11U0S0UfBPrBaGRPqktIZdTU5qwrqz/9KhW1T8T1t/YkWo6QP8I6fTTevtyFb43k+6/t0p/++XJ/EE0qvasD1fpIy0wb6ZU0+1ECtAzprvQS6PR6tPIaRq80uU4xS9qQDWm46IFpjzcHr8/m5f0+XXwOqZGMh625TBm5Y7JNDhIdRfnlOPgpciTVu781XVkKRoxvn7zYTQxti0d1z7k2Gelij4a5h/Ki3MSvFyOR6VyfOuQ+87fxrvltXQ+yA/NV38ti2VdiNvQSMaLGfw2SQP6Z/nkUdLuQu/UkYsC+utI3/5llY4E9Berz826/SVdzFnnpSbvDOJOANALQjr9su5FN3LxVVJOX9WpJgsb0bzSegLM40cf5Uj3g5fKKX9ezkEXyAA9IKTTKzYuerUa0H/5aX43eCNnT+Z3ckpfBS+jmg7QE0I6/XFRRf8weIn0mYD+5509nt8V1F9uXU0HYOeEdHpjPIhbYS76i11OcAneiqD+SpPLPQoA2CEhnf5QwXuhdQ/6Y+PxNkVQf5VsXQjAjgnp9ILdRV9MD3o3alAvYdSmR3/MLqQAOyak0xN2F/1j6b6A3p3lk/99O3LMg99J2TkJsEtCOjt3WbGbBs/LcfqX87gbdGq4Sp/U5zr4rSPjGAF2R0hn51Je3Q6eV0Jj3YTHHPTurTeDqkGd3zGOEWB3hHR2y+ZFfyylr+wkuj3LxXweOX0RPMfmRgC7I6SzU+NhuaXOb6T7Ri1u3/Kn+b1IcRI8azJ0jgLshJDOjlmc9hx96Ds1PEuflWOgxegZgzAaFWAXhHR2xoLR38uD9Jk+9N1ZtxglbS+/YQEpwA4I6eyMBaO/lY7PHs9Pgp1atxppe3mOBaQA2yeks0Ppw+BCneZyHnbA7AltL8/LKTtXAbZMSGcnhtdmt0Kry69Mc+mVeizyIN0PrtiBFGDLhHR2YhCrW8GFUkU3zaV/zs7inmr6rwYrU14AtklIZzdy+ji4UKroQf+sF/A6NldyyvYzANgiIZ2tW7e6pDAtolJF77X17PRyjIJqquUFYHuEdLZOq8uv6sjFoN/c6XhKywvA9gjp7ICpLmulQmvkYv+t73ToTV8z5QVge4R0tsoGRs9Qod0bJr08ZWMjgC0R0tmuHFpdKr3oe8Wkl1+NR3E7AOickM5WpZRvBqro+2YxX6imX0hZXzrANgjpbM9kNg1v8GvDc9vO75t0Fg+CcgrrSwfYBiGdrRkOYxYU6djuovtnuZjPI7m4KiZGMQJ0T0hna4xevJBTfBvspVVKjl0YxQiwDUI625OSfnRjF/fa+TIeWEBqFCPANgjpbMdFP7p2l5T+EeyvuoBUNb2aGcUI0C0hna3Qj35heB73gv2WLCCN2pc+ck4DdElIZytS6GEtdxLmFozuv3W7kpaXek4L6QAdEtLZiqSHtT4J3weHQduSeekAHRPS6V7tXdWPXs+24+Ag5ORYmpcO0C0hnc7pXY31VJflj/N5cBDOzmKu5SUm79YF4QB0Qkinc2klpEdKJ8HhWMwX5f+ezV90LcdaXgC6IqTTuZxWR9G4VYTRi4dmpS994AIcoDNCOp1LKX0QjRufq7oenFWcRONSxDQA6ISQTvdaXzRq9OJBWi7mzfelWzwK0B0hnU6Nrs+OonUpqaIfqJxy62M1J3YeBeiGkE6nco7m38D1ox+ulAcn0bjh0OJRgC4I6XTKTqP60Q9ZNuElBklfOkAXhHQ6lVK+GW1b6Ec/XOt56a3L0fo5DtAJIZ2utd7uIsQdsjovPbV+jPM0ANg4IZ1uNT7ZJafU+sLCw5fTo2jbNADYOCGdzowns+Y3OslZJf3gOcZTE14ANk9IpzN5ZLLLYBCnwUGzeLRckI9V0wE2TUinM8mW4bH8ca6SfuAsHo1YrYR0gE0T0ulMbn00mzaINtTFo43vPGoMI8DmCel0aDWNhqWU/hW0ofW2piykA2yakE5nSkj9azRs1fxovoaY8NL0uQ7QBSGd7jReXTPZpR05tV5Jz82vPwHYNCGdLk2jYSm13afckuZDejLJCWDThHQ68e5kNo3Gjc6MX2zFYGlWegCwUUI6nTgbedP+eTE/DZowDHdNXJgDbJaQDt04DZrhggyATRPS6UTzM9KF9BadRsPcPQPYLCGdTqTGJ7ukSP8O2tL4QmEX5gCbJaRDB3LEfwdNyTnbvAqAjRHS6UTrlfRSVRTYGpNi0HQlvfVzHmDTRgEdyLGapBJbWpVM+2hOuXuyaPcVvw7pZqX3WCkcPCjHyP+X2HfT8nEUjRDS6URK6a81tbRqZSOj9tRj3vBrvvhr0Fujs/SFKUTsu+F/zW4PVvkoGqHdBTpQKlanQVOa33UUgI0S0umKW9/QlDwNADZGSKcbjfenJu0u7Vk55gBsjpAOHRguBbbWuDADYJOEdLph0gO0ZhoAbIyQTjdS2yH9ZyMYmzM6s3AUgM0R0qELi7mQDgD8aUI6AAD0jJAOAAA9I6QDsAkWiwNskJAOwCYI6QAbJKQDAEDPCOkAANAzQjoAAPSMkA7AJpwGABsjpAMAQM8I6QAA0DNCOgAA9IyQDh14dzKbBk05NyccgA0S0unKaUBD8qjxkJ5jEQBsjJAOwNtLQjrAJgnp0IHzsdaH1uTsmAOwOUI63Wi8qiawNWjQ9jFPkf4dAGyMkE5X2g7pKaZBUwbnbYf0HPHfAcDGCOl0IuesqkZTcuOV9MI5D7BBQjqdSDFoupKeskp6cxpvccoWjgJslJBOJ3Lj7S5JT3qDVtMAgA0R0umGqtpfg6aklJo+5qWSfhoAbIyQTidaf8NOEf8ZtKbt6S5ZSAfYJCEdOpAj3wzasrIOAYDNEdLphKqanvSmTGaTcvuk6WM+OlNJB9gkIZ1OeMMuga0GN5owDlX0nxtfLA6waUI6nfCGHfGuanoz8qj5Y72IxVxIB9ggIZ1u1Dfs3HZQXw5jFjQhrRo/1trbADZOSKc7jY9hHCQtEK3IrR9rI1cBNk5Ipzutz02262hDWt/IKJ0GABslpNOdnP4VDUsWEzYjpfRBNKzcSWj6XAfogpBOZ1rf0Mis9IbktnvS7TYKsHlCOp3xxl0q6cYwHrzxZNb8AmH7IgBsnpBOZwbLmEfjxmMtL4duNXSM7YsAsHlCOp0ZmpVe2yCOgoOWklGbPy/mpwHARgnpdGb9xt34rPTy8+tLP3Ap8ofRsuyOGUAXhHS6NWh9DGO2odGha3wjo5TanuIE0BUhnW7l9ChaVlshLB49XJPZtBzjpo/vKqmkA3RBSKdbboXHaKRn+VANh45tdo4DdEJIp1MrYxjrpkZC+oEqx/YoGjdovaUNoCNCOp0an6uyJRNeDlZKjS8aLZY/zlXSATogpNMpE17WO482H+QOUl1r0PhOo9rZALojpNO9QfNv5JPR9dlRcFCGQ3dITHYB6I6QTudy6xNeQl/6IRrE6lY0Lqc4CQA6IaTTuaySXq9UPg4OTGq+jSmHcxugK0I6nRsvVduiTgExL/1gjCezemdkGo07OxPSAboipNM5i0cvDMfRfHvEwRjoR18vGl3Mmz+vAboipLMVOeXvo3GDVZjycigG2pcipebPaYAuCelsRcqDk2hevqXl5QBMZtMw+z5WoY0NoEtCOlth8ejaZDQy5WXfjY1eXLNRGUC3hHS2Yr3ATF96uaOQvwz2W8qfR+tynK7XmgDQGSGd7agLzFTTq5mWlz120eribkhKJwFAp4R0tmeV/hFMxqO4Heyl8XB1N6gsGgXomJDO1uhLv2Rjoz1mA6NqeG7RKEDXhHS25uzx/ERf+trR6PrsKNgr42uz22EDo/V8dP3oAN0T0tmupOWlGmSLD/dOyp8G5qMDbImQzlatBm6TVznCzPQ9Mp7MZmajX8gpHgQAnRPS2arzpTf4K+8M4k6wH4Yrx+rCYt22BkDnhHS2q45iTKrpVa7ztlXT+6+OXYyk1aVIodUFYFuEdLbPKMYrE9X0/jN28Vc53AkD2BYhna0brrzRX1FN7zlV9Ocsz527ANsipLN16/FtWl6uqKb3mCr6r1KUO2C1XQ2ArRDS2Ymst/Wpy2r6NOiV9UQXVfSntLoAbJeQzk6MzuI4uDIZD/M3Qb8M83fBU1pdALZLSGcntLz8jl1Ie8Tuor+VjrW6AGyXkM7umPLynJRLNd0i0t1btx7lL4OnVhHOVYAtE9LZmeUqjiOH6tyvpuPRSjjcscvFotPgQo7T8ydzrS4AWyakszvl9nlOFpA+J6c72l5256LNxWLR56R0EgBsnZDObqW4FzxH28uOaHP5Q8Pz+CoA2DohnZ06ezw/0fLyO9PxMH8dbNV4lGtLxzR41sl6kTcAWyeks3N5kO4Hv3X7P96f2eRoS0bXb3xZLhZvBr+Rvg0AdkJIZ+fOzuKeavrvrVL+Wn9698bv3/g85XQ3eF6O0+WT+XEAsBNCOrt3sYBUxe4PpJy/sxtphyazWXnt3Q1+J6miA+yUkE4/JLsZvkDdjfShoN6B8pyW5/a7VL4KfmewsiswwC4J6fTCegGpHUhfZDoerSvqwuSmXAT0h2Gh6AukYwtGAXZLSKc3ciSj3l4kx6wE9YeC+gYI6K9k7CLA7o0CeqJW08fXb9aRjEfB710G9eVk9kmocv45AvprUEU/ROfDqBulWaDPXhus8s0c7RDS6ZVaTU+Rj4I/VoP6cB3UPxLU35CA/lpqFX0ZHJ78eWop3XCQWnsJa3ehV/Smv5apxaRvpo6yHA3zDyGgv4IqOkBfCOn0jt7011KD+g/ja7PbwUtdzEHPD01xeTW96AD9IaTTO6rpr62EzvzNO+/f+DL4vclsUgL615HSveA1qKID9ImQTi+ppr++uhnP+NrNf2p/ecZkVnv3fygB/U7wWlTRAfpFSKeXVNPfWG1/+WfzVfVSPa/PwVj/+RtSRQfoGyGd/jpLXwRvpOWqel0cWsN5fQ6C15ZzLFTRAfpHSKe3lov5PEe6H7ypdVW9hPVvmgjrtbXl2s2HdXFoqJ6/sUE5x1TRAfpHSKfXzs7jbpRKX/Bn3D7osF7nnl+7cXzZ2nIUvLkcp7/8NL8bAPSOkE6/LeYloFtE+pYOKqyv21pK5bz+TOW18Wnw5yXnFkBfCen03vKn+T2LSDfiKqw/XM9Xn8z2Z274epzi7M4zbS1HwVtKx8sn8+MAoJdGAXugjmRMkY+CTTgqz+jReBjfpGs3H5Tn9h/L83hwcdeiR2owH8TtSPnjcOw3qi4WHa3iq2UA0FdCOnuhjmQcXZvdL0H982BjcsSt8vlWDexx7eZJTun78u2Ts7OYbz20l1A+GsUs5VX5O6Wbgnl30iB99fNPFosC9JmQzt6oi0hLZfXjSCZ4dOQo5XUw/rKE9ojrN+cpx2nO6fs8iHlKsVj+OJ/H2yph/N3yeTmM2SDHNKV8M19U96f1qqFEyKBD5Zgun8ztwgrQc0I6+6NUdvP12WeXPcl0LcesZOZZpHwr5fWvY3ztZn1YlMB+WqfulDhdq+3rivsqpX89+68Pcv7g1z9qfWE1LY+Tcjdkch6XC2JSXORytma4Sh9pcwHoPyGdvaLtZfdKrp7UAF+/fjZgl4un535f/uN/lx1KOX1lJjrAfjDdhb1zOTv97dsuoCE5xYmZ6AD7Q0hn/yzmi3LL/hObHMFrynE6OkufBQB7Q0hnL13csrcRC7yOlNIX2lwA9ouQzt5ab3IU+dsAXqj2of/yZP4gANgrQjp7bXk+uFNv5QfwO/rQAfaXkM5+u+hP/0h/OvyGPnSAvSaks/dqr20epE8CWMvlorVevOpDB9hfQjoHoc5Pz8lCUqjSwDx0gH0npHMwSlC/myPdD2hYXSi6fFwXVQOwz4R0DsrZk/mdSHES0KR030JRgMMgpHNwlmfrjY7sSEpTyl2kB8t6kQrAQRDSOTy/7kh6GtCCHI/eOw+TXAAOiJDOQaqL5i5HM54GHLLyGv/LKh0tysVpAHAwhHQOlqDOwSuv7foaF9ABDo+QzkET1DlYlwHdqEWAwySkc/AEdQ6OgA5w8IR0miCoczByPPrLKv1dQAc4bEI6zXgmqBvPyF7KKU4sEgVog5BOU2pQX5agniIeBOyV9O3Z40cWiQI0QkinPSXk/PLk0Sc50v2APZBy+mr5ZH47AGiGkE6zzp7M7+SUvgrotfTZLz/N7wYATRHSadrZ4/ndVax3J9VCQL/URc7n6e+lgn4cADRHSKd550/mD4ar9HeTX+iNHI/qIuflYm6RM0CjhHSIpwtK/14X5wXsVLq//OnRzIhFgLYJ6XBlMV+sF+fl9EXAluUcixTpk/IavBMANE9Ih99Y/jS/NzxPf9P+wrbU+eejcifnlydzo0EBWBPS4Q+s219+evQ301/oXEpf1Pnn2lsAeJaQDi9Rp7+UoP6Rqjobl+PRenrL4/m9AIDfENLhFUpQP1FVZ1Nq73l9LdXFoaa3APAiQjq8plpV16vO27jqPa+vpQCAlxDS4Q1c9arXXSCFdV5bea3UyS16zwF4XUI6/Al1F8i62UyOdD/gBa5aW/5icgsAb0hIhz+pVkTPnszvrFtgbILE76T7763S32pry2IxXwQAvAEhHd7SugXmyfx2Deulsv6PoGm177y+FuqmRMI5AH+WkA4bcllZv1VHNtagFjSlHvN67PWdA7AJQjpsWB3ZWINanYGtDaYF6fgqnNdjHwCwAUI6dKTOwL5qgxHWD8vTBaHn6T/LMf5MOAdg04R06NizPetGN+63dRtTTl9YEApA10YBbMVln/Jx/Rhdnx2lHLdL7Ps06LVaNU8pfVsC+oOrirlkDkDXhHTYgcuwd/LuZHb3fBhHOeXPS2ifBb1Rq+Zplf7x3iqOVcwB2DYhHXbo2ep6CezTEtjvlNLtx5FiGmzdxVSe9P17Z3HvKphL5wDsQjshPad/lU+ncfj+Heyly8B+p36MJ7NZDEqFfZA/VWHv1lXFfLiKB1ejEwXzNqSzcqiH1ogA/ZQC6LXLCvtRjlJhz3GUUkyCP+2ixzwerAbp+/+1jAdaWQDoIyEd9sx60ekqZqXK/nGUR6H95Wooj1otz+n78nyd1NGYAQA9J6TDnnvaGpPyh7WXXXvMun2hjkp89GwLCwDsEyEdDsxkMpv831GpsNdq+0VwnxxixX1dIR/EabkomddAngcxf+8s5tpXADgEQjo0olbc8ygmNbzXinuO/MF6isyqVN97GuCfC+KR/r0qQbz8fRfj85irkANwyIR0YF19/39RgnsJ8SUYTwa5fD0ogT6XIB/rCSjTq99aAvNVoJ++7p9/uVjzaYW7/Hmn9TFd7r6aU/pXuXhYrIbl95Xvjc7i9N2Ihao4AAAAAL3wP5QOtQJklKNqAAAAAElFTkSuQmCC"/>
</defs>
</svg>
