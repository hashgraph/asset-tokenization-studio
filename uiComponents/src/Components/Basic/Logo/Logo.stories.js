var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Logo } from "./Logo";
var meta = {
    title: "Design System/Basic/Logo",
    component: Logo,
    argTypes: {
        variant: {
            control: false,
            description: "The variant of the Logo. Must be defined in the theme"
        },
        size: {
            options: ["full", "iso"],
            control: { type: "select" },
            description: "Size of the logo. Can be 'full' or 'iso'"
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1826-13793&t=3ozoVrWQYvgw1fj3-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Logo, __assign({}, args));
};
export var Full = Template.bind({});
Full.args = {
    size: "full"
};
export var Iso = Template.bind({});
Iso.args = {
    size: "iso"
};
