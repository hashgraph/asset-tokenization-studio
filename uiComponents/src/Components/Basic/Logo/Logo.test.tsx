import { render } from "@/test-utils";
import React from "react";
import type { LogoProps } from "./Logo";
import { Logo } from "./Logo";

describe(`<Logo />`, () => {
  const defaultProps = {
    width: "100%",
    height: "100px",
    alt: "logo",
  };
  const factoryComponent = (props: LogoProps = defaultProps) =>
    render(<Logo {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.asFragment()).toMatchSnapshot("SizeFull");
  });

  test("renders correctly size iso", () => {
    const component = factoryComponent({ ...defaultProps, size: "iso" });
    expect(component.asFragment()).toMatchSnapshot("SizeISO");
  });
});
