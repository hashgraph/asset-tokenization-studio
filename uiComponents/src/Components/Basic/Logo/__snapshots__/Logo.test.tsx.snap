// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders correctly size iso: SizeISO 1`] = `
<DocumentFragment>
  <img
    alt="logo"
    src="[object Object]"
    style="width: 100%; height: 100px;"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Logo /> renders correctly: SizeFull 1`] = `
<DocumentFragment>
  <img
    alt="logo"
    src="[object Object]"
    style="width: 100%; height: 100px;"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
